data-processing/entities/sentences/extractor.py

```diff
@@ -45,16 +45,20 @@ def parse(self, tex_path: str, tex: str) -> Iterator[Sentence]:
 
         if last_segment is not None:
             plaintext_to_tex_offset_map[len(plaintext)] = last_segment.tex_end
-
         # Segment the plaintext. Return offsets for each setence relative to the TeX input
         segmenter = pysbd.Segmenter(language="en", clean=False, char_span=True)
+        # Record the current length of the plain text so account for the extractor bug
+        length_so_far_in_plain_text = -1
         for i, sentence in enumerate(segmenter.segment(plaintext)):
+            # Since the sentence extractor has several bugs related to finding start and end indicies
+            # we will simply set them ourselves.
+            start = plaintext_to_tex_offset_map[length_so_far_in_plain_text + 1]
+            end = plaintext_to_tex_offset_map[sentence.end + (length_so_far_in_plain_text + 1 - sentence.start)]
+            length_so_far_in_plain_text += sentence.end - sentence.start + 1
```

Comment:
@zkirby I like the simplicity of the solution. Can you guarantee that these length computations will be correct? A couple cases that I expect could confuse this computation are:

* What if a sentence ends with a bunch of spaces (`Hello there     .`), does `.end` still end at the period?
* What if there is punctuation between sentences that's more than one character long (`Hello there!?`)

Can you take a look at the `pysbd` code to the extent that we know this is a robust solution?

Reply:
Hmmm there are definitely still sentences that break pysbd. In particular having multiple punctuation will still break it pysbd but if that is something we want to address it should probably be done in another PR since it's an issue with pysbd and not this particular solution.

I have noticed however that there is definitely a white space bug with this solution that I will investigate.

Reply:
Roger, thanks for looking into this!